data/store/mongo/mongo_data_set.go

```diff
@@ -68,25 +66,33 @@ func (d *DataSetRepository) EnsureIndexes() error {
 				{Key: "_active", Value: 1},
 			},
 			Options: options.Index().
-				SetBackground(true).
 				SetName("UploadId"),
 		},
+		{
+			Keys: bson.D{
+				{Key: "_userId", Value: 1},
+				{Key: "client.name", Value: 1},
+				{Key: "type", Value: 1},
+				{Key: "createdTime", Value: -1},
+			},
+			Options: options.Index().
+				SetName("ListUserDataSets").
+				SetPartialFilterExpression(bson.D{
+					{Key: "_active", Value: true},
+				}),
+		},
 		{
 			Keys: bson.D{
 				{Key: "_userId", Value: 1},
 				{Key: "deviceId", Value: 1},
 				{Key: "type", Value: 1},
-				{Key: "_active", Value: 1},
-				{Key: "_deduplicator.hash", Value: 1},
+				{Key: "createdTime", Value: -1},
 			},
 			Options: options.Index().
-				SetBackground(true).
+				SetName("ListUserDataSetsAlso").
```

Comment:
Kind of confused about this name, but otherwise looks fine.

Reply:
been a while, but i believe both are used by listing user datasets in different ways, suggestions?

Reply:
Perhaps `ListUserDataSetsByDeviceId` or `ListUserDataSetsDeviceId` or `DataSetsByDeviceId` ? Naming is hard...

Reply:
replaced with option 2